spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/pubsub/GcpPubSubEmulatorConfiguration.java

```diff
@@ -0,0 +1,62 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.autoconfigure.pubsub;
+
+import com.google.api.gax.core.CredentialsProvider;
+import com.google.api.gax.core.NoCredentialsProvider;
+import com.google.api.gax.grpc.GrpcTransportChannel;
+import com.google.api.gax.rpc.FixedTransportChannelProvider;
+import com.google.api.gax.rpc.TransportChannelProvider;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+
+import org.springframework.boot.autoconfigure.AutoConfigureBefore;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.gcp.autoconfigure.core.GcpContextAutoConfiguration;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * If <code>spring.cloud.gcp.pubsub.emulatorHost</code> is set, spring stream will connect
+ * to a running pub/sub emulator.
+ *
+ * @author Andreas Berger
+ */
+@Configuration
+@ConditionalOnProperty("spring.cloud.gcp.pubsub.emulatorHost")
```

Comment:
Consider to use `perfix` in this annotation - better performance during parsing and resolution.

And the time I'm still asking: shouldn't we have a special `Conditional` to be sure that emulator is really started on that `emulatorHost`? I can configure the property or may forget that I have it in the ENV var, but that doesn't mean yet that the real emulator is alive.

Thanks

Reply:
If the property is set, the user needs to ensure that the emulator is running. If I make this configuration `@Conditional` on the open port of the emulator, the app could accidentially connect to the real pub/sub, which should be prevented.

Reply:
Yeah... Right. Never mind.

We definitely don't check the connection in any other configuration.
So, yes, we can go ahead just with the property!

👍